### PR TITLE
Add docs for Postgres/MySQL Azure support for DBM

### PR DESCRIPTION
**Pull request by jmeunier28**
### What does this PR do?

Adds documentation for Database Monitoring setup for Azure MySQL/PostgreSQL managed database in prep for GA'ing full support. Currently, we will only be supporting `Single Server` deployment type.  

### Preview

[postgres](https://docs-staging.datadoghq.com/jmeunier/DBM-azure-pg-and-mysql/database_monitoring/setup_postgres/azure/)
[mysql](https://docs-staging.datadoghq.com/jmeunier/DBM-azure-pg-and-mysql/database_monitoring/setup_mysql/azure/)

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: layouts/partials/dbm/dbm-setup-postgres.html [hunk 1]
```diff
@@ -23,6 +23,11 @@
           {{ partial "img.html" (dict "root" . "src" "integrations_logos/google_cloudsql.png" "class" "img-fluid" "alt" "Google Cloud SQL" "width" "175") }}
         </div>
       </a>
+      <a class="card" href="/database_monitoring/setup_postgres/azure">
+        <div class="card-body text-center py-2 px-1">
+          {{ partial "img.html" (dict "root" . "src" "integrations_logos/azure_db_for_postgresql.png" "class" "img-fluid" "alt" "PostgreSQL" "width" "175") }}
+        </div>
```
Comment (jmeunier28) [inline layouts/partials/dbm/dbm-setup-postgres.html:29]: Screen Shot 2022-04-01 at 2 34 08 PM
 
 I added this to the `Setting up Postgres` page, but Azure is not showing up a link in the side panel. Not sure where to add it?

Comment (jtappa) [reply layouts/partials/dbm/dbm-setup-postgres.html:29]: The sidenav items get added manually in menus.yaml files, usually we talk about this when you're planning to add new doc pages, and your docs writer takes care of it. I'll add these for you.

File: content/en/database_monitoring/setup_mysql/azure.md [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+
+---
+
+{{< site-region region="us5,gov" >}}
+<div class="alert alert-warning">Database Monitoring is not supported for this site.</div>
+{{< /site-region >}}
+
+Database Monitoring provides deep visibility into your MySQL databases by exposing query metrics, query samples, explain plans, connection data, system metrics, and telemetry for the InnoDB storage engine.
+
+The Agent collects telemetry directly from the database by logging in as a read-only user. Do the following setup to enable Database Monitoring with your MySQL database:
```
Comment (jtappa) [inline content/en/database_monitoring/setup_mysql/azure.md:18]:
```suggestion
The Agent collects telemetry directly from the database by logging in as a read-only user. Complete the following steps to enable Database Monitoring with your MySQL database:
```

File: content/en/database_monitoring/setup_mysql/azure.md [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+{{% /tab %}}
+{{% tab "Docker" %}}
+
+To configure the Database Monitoring Agent running in a Docker container, you can set the [Autodiscovery Integration Templates][1] as Docker labels on your agent container.
+
+**Note**: The Agent must have read permission on the Docker socket for Autodiscovery of labels to work.
+
+### Command line
+
+Get up and running quickly by executing the following command to run the agent from your command line. Replace the values to match your account and environment:
```
Comment (jtappa) [inline content/en/database_monitoring/setup_mysql/azure.md:170]:
```suggestion
Execute the following command to run the agent from your command line. Replace the values to match your account and environment:
```

File: content/en/database_monitoring/setup_mysql/azure.md [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+
+[1]: /agent/docker/integrations/?tab=docker
+[2]: /agent/guide/secrets-management
+[3]: /agent/faq/template_variables/
+{{% /tab %}}
+{{% tab "Kubernetes" %}}
+
+If you have a Kubernetes cluster, use the [Datadog Cluster Agent][1] for Database Monitoring.
+
+Follow the instructions to [enable the cluster checks][2] if not already enabled in your Kubernetes cluster. You can declare the MySQL configuration either with static files mounted in the Cluster Agent container or using service annotations:
```
Comment (jtappa) [inline content/en/database_monitoring/setup_mysql/azure.md:213]:
```suggestion
Follow the instructions to [enable the cluster checks][2] if not already enabled in your Kubernetes cluster. You can declare the MySQL configuration with static files mounted in the Cluster Agent container, or by using service annotations:
```

File: content/en/database_monitoring/setup_mysql/azure.md [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+      ]
+spec:
+  ports:
+  - port: 3306
+    protocol: TCP
+    targetPort: 3306
+    name: mysql
+```
+
+The Cluster Agent automatically registers this configuration and begin running the MySQL check.
```
Comment (jtappa) [inline content/en/database_monitoring/setup_mysql/azure.md:286]:
```suggestion
The Cluster Agent automatically registers this configuration and begins running the MySQL check.
```

File: content/en/database_monitoring/setup_mysql/azure.md [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+[1]: /agent/cluster_agent
+[2]: /agent/cluster_agent/clusterchecks/
+[3]: https://helm.sh
+[4]: /agent/guide/secrets-management
+{{% /tab %}}
+{{< /tabs >}}
+
+### Validate
+
+[Run the Agent's status subcommand][6] and look for `mysql` under the Checks section. Or visit the [Databases][7] page to get started!
```
Comment (jtappa) [inline content/en/database_monitoring/setup_mysql/azure.md:299]:
```suggestion
[Run the Agent's status subcommand][6] and look for `mysql` under the **Checks** section. Or visit the [Databases][7] page to get started.
```

File: content/en/database_monitoring/setup_postgres/aurora.md [hunk 7]
```diff
@@ -45,7 +45,7 @@ Configure the following [parameters][3] in the [DB parameter group][4] and then
 
 | Parameter | Value | Description |
 | --- | --- | --- |
-| `shared_preload_libraries` | `pg_stat_statements` | Required for `postgresql.queries.*` metrics. Enables collection of query metrics via the the [pg_stat_statements][5] extension. On by default in Aurora. |
+| `shared_preload_libraries` | `pg_stat_statements` | Required for `postgresql.queries.*` metrics. Enables collection of query metrics via the [pg_stat_statements][5] extension. On by default in Aurora. |
```
Comment (jtappa) [inline content/en/database_monitoring/setup_postgres/aurora.md:48]:
```suggestion
| `shared_preload_libraries` | `pg_stat_statements` | Required for `postgresql.queries.*` metrics. Enables collection of query metrics using the [pg_stat_statements][5] extension. On by default in Aurora. |
```

File: content/en/database_monitoring/setup_postgres/azure.md [hunk 8]
```diff
@@ -0,0 +1,416 @@
...
+
+Proxies, load balancers, and connection poolers
+: The Agent must connect directly to the host being monitored. For self-hosted databases, `127.0.0.1` or the socket is preferred. The Agent should not connect to the database through a proxy, load balancer, or connection pooler such as `pgbouncer`. While this can be an anti-pattern for client applications, each Agent must have knowledge of the underlying hostname and should stick to a single host for its lifetime, even in cases of failover. If the Datadog Agent connects to different hosts while it is running, the values of metrics will be incorrect.
+
+Data security considerations
+: See [Sensitive information][2] for information about what data the Agent collects from your databases and how to ensure it is secure.
+
+## Configure Postgres settings
+
+Configure the following [parameters][3] in the [Server parameters][4] and then **restart the server** for the settings to take effect.
```
Comment (jtappa) [inline content/en/database_monitoring/setup_postgres/azure.md:47]:
```suggestion
Configure the following [parameters][3] in the [Server parameters][4], then **restart the server** for the settings to take effect.
```

File: content/en/database_monitoring/setup_postgres/azure.md [hunk 9]
```diff
@@ -0,0 +1,416 @@
...
+
+{{< tabs >}}
+{{% tab "Single Server" %}}
+
+| Parameter | Value | Description |
+| --- | --- | --- |
+| `track_activity_query_size` | `4096` | Required for collection of larger queries. Increases the size of SQL text in `pg_stat_activity` and `pg_stat_statements`. If left at the default value then queries longer than `1024` characters will not be collected. |
```
Comment (jtappa) [inline content/en/database_monitoring/setup_postgres/azure.md:54]:
```suggestion
| `track_activity_query_size` | `4096` | Required for collection of larger queries. Increases the size of SQL text in `pg_stat_activity` and `pg_stat_statements`. If left at the default value, queries longer than `1024` characters are not collected. |
```

File: content/en/database_monitoring/setup_postgres/azure.md [hunk 10]
```diff
@@ -0,0 +1,416 @@
...
+| `pg_stat_statements.track` | `ALL` | Optional. Enables tracking of statements within stored procedures and functions. |
+| `pg_stat_statements.max` | `10000` | Optional. Increases the number of normalized queries tracked in `pg_stat_statements`. This setting is recommended for high-volume databases that see many different types of queries from many different clients. |
+
+{{% /tab %}}
+{{% tab "Flexible Server" %}}
+
+| Parameter            | Value | Description |
+|----------------------| -- | --- |
+| `azure.extensions` | `pg_stat_statements` | Required for `postgresql.queries.*` metrics. Enables collection of query metrics via the [pg_stat_statements][1] extension. |
```
Comment (jtappa) [inline content/en/database_monitoring/setup_postgres/azure.md:63]:
```suggestion
| `azure.extensions` | `pg_stat_statements` | Required for `postgresql.queries.*` metrics. Enables collection of query metrics using the [pg_stat_statements][1] extension. |
```

File: content/en/database_monitoring/setup_postgres/azure.md [hunk 11]
```diff
@@ -0,0 +1,416 @@
...
+| `track_activity_query_size` | `4096` | Required for collection of larger queries. Increases the size of SQL text in `pg_stat_activity` and `pg_stat_statements`. If left at the default value then queries longer than `1024` characters will not be collected. |
```
Comment (jtappa) [inline content/en/database_monitoring/setup_postgres/azure.md:64]:
```suggestion
| `track_activity_query_size` | `4096` | Required for collection of larger queries. Increases the size of SQL text in `pg_stat_activity` and `pg_stat_statements`. If left at the default value, queries longer than `1024` characters are not collected. |
```

File: content/en/database_monitoring/setup_postgres/azure.md [hunk 12]
```diff
@@ -0,0 +1,416 @@
...
+
+[1]: https://www.postgresql.org/docs/current/pgstatstatements.html
+{{% /tab %}}
+{{< /tabs >}}
+
+## Grant the Agent access
+
+The Datadog Agent requires read-only access to the database server in order to collect statistics and queries.
+
+Choose a PostgreSQL database on the database server to which the Agent will connect. The Agent can collect telemetry from all databases on the database server regardless of which one it connects to, so a good option is to use the default `postgres` database. Choose a different database only if you need the Agent to run [custom queries against data unique to that database][5].
```
Comment (jtappa) [inline content/en/database_monitoring/setup_postgres/azure.md:76]:
```suggestion
Choose a PostgreSQL database on the database server the Agent to connect to. The Agent can collect telemetry from all databases on the database server regardless of which one it connects to, so a good option is to use the default `postgres` database. Choose a different database only if you need the Agent to run [custom queries against data unique to that database][5].
```

File: content/en/database_monitoring/setup_postgres/azure.md [hunk 13]
```diff
@@ -0,0 +1,416 @@
...
+CREATE OR REPLACE FUNCTION datadog.pg_stat_statements() RETURNS SETOF pg_stat_statements AS
+    $$ SELECT * FROM pg_stat_statements; $$
+LANGUAGE sql
+SECURITY DEFINER;
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+**Note**: When generating custom metrics that require querying additional tables, you may need to grant the `SELECT` permission on those tables to the `datadog` user. Example: `grant SELECT on <TABLE_NAME> to datadog;`. See [PostgreSQL custom metric collection explained][5] for more information.
```
Comment (jtappa) [inline content/en/database_monitoring/setup_postgres/azure.md:130]:
```suggestion
**Note**: When generating custom metrics that require querying additional tables, you may need to grant the `SELECT` permission on those tables to the `datadog` user. Example: `grant SELECT on <TABLE_NAME> to datadog;`. See  the explanation of [PostgreSQL custom metric collection][5] for more information.
```

File: content/en/database_monitoring/setup_postgres/azure.md [hunk 14]
```diff
@@ -0,0 +1,416 @@
...
+## Install the Agent
+
+To monitor Azure Postgres databases, install the Datadog Agent in your infrastructure and configure it to connect to each instance endpoint remotely. The Agent does not need to run on the database, it only needs to connect to it. For additional Agent installation methods not mentioned here, see the [Agent installation instructions][7].
+
+{{< tabs >}}
+{{% tab "Host" %}}
+
+To configure collecting Database Monitoring metrics for an Agent running on a host, for example when you provision a small virtual machine for the Agent to collect from an Azure database:
+
+1. Edit the `postgres.d/conf.yaml` file to point to your `host` / `port` and set the masters to monitor. See the [sample postgres.d/conf.yaml][1] for all available configuration options.
```
Comment (jtappa) [inline content/en/database_monitoring/setup_postgres/azure.md:202]: Can we replace "masters" here with something else, or is this a term postgres is using specifically?

File: content/en/database_monitoring/setup_postgres/azure.md [hunk 15]
```diff
@@ -0,0 +1,416 @@
...
+{{% /tab %}}
+{{% tab "Docker" %}}
+
+To configure the Database Monitoring Agent running in a Docker container, you can set the [Autodiscovery Integration Templates][1] as Docker labels on your agent container.
+
+**Note**: The Agent must have read permission on the Docker socket for Autodiscovery of labels to work.
+
+### Command line
+
+Get up and running quickly by executing the following command to run the agent from your command line. Replace the values to match your account and environment:
```
Comment (jtappa) [inline content/en/database_monitoring/setup_postgres/azure.md:231]:
```suggestion
Execute the following command to run the Agent from your command line. Replace the values to match your account and environment:
```

File: content/en/database_monitoring/setup_postgres/azure.md [hunk 16]
```diff
@@ -0,0 +1,416 @@
...
+For Postgres 9.6, add the following settings to the instance config where host and port are specified:
+
+```yaml
+pg_stat_statements_view: datadog.pg_stat_statements()
+pg_stat_activity_view: datadog.pg_stat_activity()
+```
+
+### Dockerfile
+
+Labels can also be specified in a `Dockerfile`, so you can build and deploy a custom agent without changing any infrastructure configuration:
```
Comment (jtappa) [inline content/en/database_monitoring/setup_postgres/azure.md:260]:
```suggestion
Labels can also be specified in a `Dockerfile`, so you can build and deploy a custom Agent without changing any infrastructure configuration:
```

File: content/en/database_monitoring/setup_postgres/azure.md [hunk 17]
```diff
@@ -0,0 +1,416 @@
...
+
+[1]: /agent/docker/integrations/?tab=docker
+[2]: /agent/guide/secrets-management
+[3]: /agent/faq/template_variables/
+{{% /tab %}}
+{{% tab "Kubernetes" %}}
+
+If you have a Kubernetes cluster, use the [Datadog Cluster Agent][1] for Database Monitoring.
+
+Follow the instructions to [enable the cluster checks][2] if not already enabled in your Kubernetes cluster. You can declare the Postgres configuration either with static files mounted in the Cluster Agent container or using service annotations:
```
Comment (jtappa) [inline content/en/database_monitoring/setup_postgres/azure.md:288]:
```suggestion
Follow the instructions to [enable the cluster checks][2] if not already enabled in your Kubernetes cluster. You can declare the Postgres configuration with static files mounted in the Cluster Agent container, or using service annotations:
```

File: content/en/database_monitoring/setup_postgres/azure.md [hunk 18]
```diff
@@ -0,0 +1,416 @@
...
+```
+
+For Postgres 9.6, add the following settings to the instance config where host and port are specified:
+
+```yaml
+pg_stat_statements_view: datadog.pg_stat_statements()
+pg_stat_activity_view: datadog.pg_stat_activity()
+```
+
+The Cluster Agent automatically registers this configuration and begin running the Postgres check.
```
Comment (jtappa) [inline content/en/database_monitoring/setup_postgres/azure.md:378]:
```suggestion
The Cluster Agent automatically registers this configuration and begins running the Postgres check.
```

File: content/en/database_monitoring/setup_postgres/azure.md [hunk 19]
```diff
@@ -0,0 +1,416 @@
...
+{{% /tab %}}
+{{< /tabs >}}
+
+### Validate
+
+[Run the Agent's status subcommand][8] and look for `postgres` under the Checks section. Or visit the [Databases][9] page to get started!
+
+## Install the Azure PostgreSQL Integration
+
+To collect more comprehensive database metrics from AZURE, install the [Azure PostgreSQL integration][10] (optional).
```
Comment (jtappa) [inline content/en/database_monitoring/setup_postgres/azure.md:395]:
```suggestion
To collect more comprehensive database metrics from Azure, install the [Azure PostgreSQL integration][10] (optional).
```

File: content/en/database_monitoring/setup_postgres/rds.md [hunk 20]
```diff
@@ -45,7 +45,7 @@ Configure the following [parameters][3] in the [DB parameter group][4] and then
 
 | Parameter | Value | Description |
 | --- | --- | --- |
-| `shared_preload_libraries` | `pg_stat_statements` | Required for `postgresql.queries.*` metrics. Enables collection of query metrics via the the [pg_stat_statements][5] extension. |
+| `shared_preload_libraries` | `pg_stat_statements` | Required for `postgresql.queries.*` metrics. Enables collection of query metrics via the [pg_stat_statements][5] extension. |
```
Comment (jtappa) [inline content/en/database_monitoring/setup_postgres/rds.md:48]:
```suggestion
| `shared_preload_libraries` | `pg_stat_statements` | Required for `postgresql.queries.*` metrics. Enables collection of query metrics using the [pg_stat_statements][5] extension. |
```

File: content/en/database_monitoring/setup_postgres/selfhosted.md [hunk 21]
```diff
@@ -48,7 +48,7 @@ Configure the following [parameters][4] in the `postgresql.conf` file and then *
 
 | Parameter | Value | Description |
 | --- | --- | --- |
-| `shared_preload_libraries` | `pg_stat_statements` | Required for `postgresql.queries.*` metrics. Enables collection of query metrics via the the [pg_stat_statements][5] extension. |
+| `shared_preload_libraries` | `pg_stat_statements` | Required for `postgresql.queries.*` metrics. Enables collection of query metrics via the [pg_stat_statements][5] extension. |
```
Comment (jtappa) [inline content/en/database_monitoring/setup_postgres/selfhosted.md:51]:
```suggestion
| `shared_preload_libraries` | `pg_stat_statements` | Required for `postgresql.queries.*` metrics. Enables collection of query metrics using the [pg_stat_statements][5] extension. |
```

File: content/en/database_monitoring/setup_mysql/azure.md [hunk 22]
```diff
@@ -0,0 +1,321 @@
...
+
+[1]: /agent/basic_agent_usage#agent-overhead
+[2]: /database_monitoring/data_collected/#sensitive-information
+[3]: https://docs.microsoft.com/en-us/azure/mysql/howto-server-parameters
+[4]: https://dev.mysql.com/doc/refman/8.0/en/creating-accounts.html
+[5]: https://app.datadoghq.com/account/settings#agent
+[6]: /agent/guide/agent-commands/#agent-status-and-information
+[7]: https://app.datadoghq.com/databases
+[8]: /integrations/azure_db_for_mysql
+[9]: /database_monitoring/troubleshooting/?tab=mysql
```
Comment (jtappa) [inline content/en/database_monitoring/setup_mysql/azure.md:321]:
```suggestion
[9]: /database_monitoring//setup_mysql/troubleshooting
```

File: content/en/database_monitoring/setup_postgres/azure.md [hunk 23]
```diff
@@ -0,0 +1,416 @@
...
+[2]: /database_monitoring/data_collected/#sensitive-information
+[3]: https://www.postgresql.org/docs/current/config-setting.html
+[4]: https://docs.microsoft.com/en-us/azure/postgresql/howto-configure-server-parameters-using-portal
+[5]: /integrations/faq/postgres-custom-metric-collection-explained/
+[6]: https://www.postgresql.org/docs/current/app-psql.html
+[7]: https://app.datadoghq.com/account/settings#agent
+[8]: /agent/guide/agent-commands/#agent-status-and-information
+[9]: https://app.datadoghq.com/databases
+[10]: /integrations/azure_db_for_postgresql/
+[11]: /database_monitoring/troubleshooting/?tab=postgres
```
Comment (jtappa) [inline content/en/database_monitoring/setup_postgres/azure.md:416]:
```suggestion
[11]: /database_monitoring/setup_postres/troubleshooting
```

**Review by jtappa — CHANGES_REQUESTED** [review]

File: content/en/database_monitoring/setup_mysql/azure.md (hunk 22 above)
Comment (jmeunier28) [reply content/en/database_monitoring/setup_mysql/azure.md:321]: looks like you added an extra `/` here, ill fix after

**Comment by jmeunier28** [comment]
@jtappa thanks for the review! I think I got everything

**Comment by jtappa** [comment]
Just trying to get the preview to work after yesterday's build failures.

**Comment by jmeunier28** [comment]
@jtappa awesome, looks like the build preview is working now. I sent out of my team to get one final look

**Comment by jtappa** [comment]
Ok, once the next build preview is up and everything looks good, I'll merge this PR, and open a separate one that can be approved and merged when we are ready for the docs to be more publicly shown. 🤞🏻

**Review by jtappa — APPROVED** [review]

**Pull request merged**